macros/edr/tests/on_run_end/handle_tests_results.sql

```diff
@@ -1,373 +1,44 @@
-        'database_name': database_name,
-        'schema_name': schema_name,
-        'table_name': table_name,
-        'column_name': column_name,
-        'test_type': 'anomaly_detection',
-        'test_sub_type': metric_name,
-        'test_results_description': test_results_description,
-        'other': elementary.insensitive_get_dict_value(anomaly_dict, 'anomalous_value'),
-        'owners': elementary.insensitive_get_dict_value(test_node, 'model_owners'),
-        'tags': elementary.insensitive_get_dict_value(test_node, 'model_tags'),
-        'test_results_query': test_results_query,
-        'test_name': elementary.insensitive_get_dict_value(test_node, 'name'),
-        'test_params': elementary.insensitive_get_dict_value(test_node, 'test_params'),
-        'severity': elementary.insensitive_get_dict_value(test_node, 'severity'),
-        'status': elementary.insensitive_get_dict_value(run_result_dict, 'status'),
-        'failures': elementary.insensitive_get_dict_value(run_result_dict, 'failures'),
-        'test_short_name': elementary.insensitive_get_dict_value(test_node, 'short_name'),
-        'test_alias': elementary.insensitive_get_dict_value(test_node, 'alias')
-    } %}
-    {{ return(test_result_dict) }}
-{% endmacro %}
+  {% for result in results %}
+    {% set result = result.to_dict() %}
+    {% set elementary_test_results_rows = cached_elementary_test_results.get(result.node.unique_id) %}
 
-{% macro get_dimension_metric_test_result(run_result_dict, test_node) %}
-    {% set test_anomaly_scores_table = elementary.get_elementary_test_table(test_node.name, 'anomaly_scores') %}
-    {% set anomalous_dimensions = [] %}
-    {% if run_result_dict.get('status') == 'pass' %}
-        {% set most_recent_anomalies_scores = elementary.get_most_recent_anomaly_scores(test_anomaly_scores_table) %}
-    {% else %}
-        {% set most_recent_anomalies_scores = elementary.get_test_result_rows_as_dicts(test_node) %}
-        {% for anomaly in most_recent_anomalies_scores %}
-            {% set anomaly_dimension = elementary.insensitive_get_dict_value(anomaly, 'dimension_value') %}
-            {% if anomaly_dimension %}
-                {% do anomalous_dimensions.append(anomaly_dimension) %}      
-            {% endif %}
-        {% endfor %}
-    {% endif %}
-    {% set recent_anomaly_sample = most_recent_anomalies_scores[0] if most_recent_anomalies_scores else {} %}
-    {% if anomalous_dimensions %}
-        {% set test_results_description = "There are " ~ anomalous_dimensions | length ~ " anomalous values for '" ~ recent_anomaly_sample.get('dimension') ~ "'" %}
-    {% else %}
-        {% set test_results_description = none %}
+    {# Materializing the test failed and therefore was not added to the cache. #}
+    {% if not elementary_test_results_rows %}
+      {% set flattened_test = elementary.flatten_test(result.node) %}
+      {% set elementary_test_results_rows = [elementary.get_dbt_test_result_row(flattened_test)] %}
```

Comment:
mmmm how is it going to work on tests that are not dbt_test?
How is it going to affect the report when an anomaly test returns results in the format of a dbt test?